src/components/navBar/index.tsx

```diff
@@ -1,17 +1,94 @@
-import Image from 'next/image';
-import styles from '@/components/navBar/navBar.module.scss';
+import { useEffect, useState } from "react";
+import Image from "next/image";
+import userData from "@/helperFunctions/getUser";
+import { toast, ToastTypes } from "@/helperFunctions/toast";
+import navLinks from "./navLinks";
+import NavbarLinks from "./NavbarLinks";
+import UserSignIn from "./UserSignIn";
+import {
+  RDS_LOGO,
+  HAMBURGER_ICON,
+  SIGN_IN_WITH_GITHUB,
+  SIGN_IN,
+  HOME_PAGE_LINK,
+  TWENTY_FIVE_PX,
+  THIRTY_PX,
+  FIFTY_PX,
+} from "@/components/constants/navbar";
+import styles from "@/components/navBar/navBar.module.scss";
 
-const RDSLogo = '/RDSLogo.png';
+const NavBar = () => {
+  const [toggle, setToggle] = useState(false);
+  const [userIsLoggedIn, setUserIsLoggedIn] = useState(false);
+  const [userDetails, setUserDetails] = useState({
+    userName: "",
+    firstName: "",
+    profilePicture: "",
+  });
 
-const NavBar = () => (
-  <nav className={styles.navBar}>
-    <a className={styles.logo} href="https://realdevsquad.com"><Image width="45px" height="45px" src={RDSLogo} alt="real-dev squad" /></a>
-    <a href="https://welcome.realdevsquad.com/">Welcome</a>
-    <a href="https://www.realdevsquad.com/events.html">Events</a>
-    <a href="https://members.realdevsquad.com/">Members</a>
-    <a href="https://crypto.realdevsquad.com/">Crypto</a>
-    <a className={styles.active} href="https://status.realdevsquad.com/">Status</a>
-  </nav>
-);
+  useEffect(() => {
+    (async () => {
+      await userData()
+        .then((data) => {
+          const { username, first_name, picture } = data;
+          setUserDetails({
+            userName: username,
+            firstName: first_name,
+            profilePicture: picture?.url,
+          });
+
+          setUserIsLoggedIn(true);
+        })
+        .catch((error) => {
+          toast(ToastTypes.ERROR, (error as Error).message);
```

Comment:
In the case of a user who is not signed in, three toast errors are being displayed with one of them being a TypeError. 
Should'nt we show a single error message or no error?
<img width="318" alt="image" src="https://user-images.githubusercontent.com/46373689/174427522-9e2cb4ba-8e7f-4152-a649-ad0efc85b3d9.png">

Reply:
Is this in `local host` or `dev.realdevsquad.com`

Reply:
> Is this in `local host` or `dev.realdevsquad.com` ??

In `dev.realdevsquad.com`

Reply:
Thanks, will handle the error